tsk/img/img_types.c

```
@@ -94,6 +98,10 @@ tsk_img_type_toid_utf8(const char *str)
 TSK_IMG_TYPE_ENUM
 tsk_img_type_toid(const TSK_TCHAR * str)
 {
+    if (str == NULL) {
+        return TSK_IMG_TYPE_UNSUPP;
+    }
+    
```

Please use `nullptr`. 

I think that it should throw an exception or return a different error. This is calling error, not an unsupported filetype. My preference is to throw an exception, but we do not have exceptions widely implemented in TSK. However, this is an okay time to start.

---

img_types is written in C so there are a couple of issues because of language incompatibility:
* There is no ``nullptr`` keyword. 
* Exceptions aren't natively supported.

 How should I proceed?

---

Well, `TSK_IMG_TYPE_UNSUPP` is wrong, because `NULL` is not a type. See if there is a return code for null pointer inappropriately provided.

---

I couldn't find a return type for a null pointer specifically. I was thinking that I could use ``TSK_IMG_TYPE_DETECT = 0x0000``. Otherwise I could ``exit(1)``.

---

Hm. I'm now aware that you are adding code to `img_types.c` rather than adding a test case. You are making the code more robust, which is good, but this is not strictly improving the code coverage.  How about just making this an `assert` statement? Just assert it is not NULL. 

You may find this article useful: https://cacm.acm.org/research/mars-code/